### ISSUE TITLE
Build for linux_aarch64?

### ISSUE THREAD

**Issue by stuarteberg** [conda-forge/graph-tool-feedstock#96]
Out of curiosity, I did a quick investigation of what it would take to build this feedstock for the `linux-aarch64` platform, which is compatible with Raspberry Pi and also Amazon's Graviton processors, IIUC.

As far as I can tell, there are only two dependencies which DON'T already have `linux-aarch64` packages available:

- `sparsehash`
- `cairomm`  (note: `cairo` and `pycairo` are already supported)

And *their* dependencies are already available on `linux-aarch64`.  So in principle it might not take much to upgrade those two.  Then we'd have to work on the `graph-tool` feedstock itself, of course.

(I suspect the story is similar for `linux-ppc64le`, but I didn't actually check.)

If we're lucky, it could be as simple as [adding the following lines to `conda-forge.yml`][1] in the `sparsehash`, `cairomm`, and `graph-tool` feedstocks:

```yaml
provider:
  linux_ppc64le: default
  linux_aarch64: default
```

[1]: https://conda-forge.org/docs/maintainer/conda_forge_yml.html#provider

Personally, I do not use the `linux-aarch64` platform for my own work, so I'm not highly motivated to add this support.  But it might not be too hard... On the other hand, wading through build errors is nobody's idea of a good time.

BTW, I notice that two somewhat analogous packages, `igraph` and `networkit`, are already available for `linux-aarch64`, with modest download counts for that platform ([see here][2] and [here][3]).  One can speculate about whether the `graph-tool` download counts for that platform would be similar, were it available.  That might help us gauge the impact of working on this.

[2]: https://anaconda.org/conda-forge/python-igraph/files
[3]: https://anaconda.org/conda-forge/networkit/files

**Comment by count0** [comment]
Sparsehash is header only, so I see no issue. I think cairomm has a compiled component, but it's trivial.

Having graph-tool on the raspberry pi would be quite nice, specially since its difficult to compile it by hand on those devices.

I say we should give it a try.

**Comment by count0** [comment]
I added a MR for both projects... let's see.

**Comment by count0** [comment]
Out of curiosity: how many dependencies are available on windows?

**Comment by stuarteberg** [comment]
>Out of curiosity: how many dependencies are available on windows?

I didn't check.  I don't know the incantation to get conda to tell me which dependencies are already available, so I manually checked anaconda.org for the `aarch64` dependencies.  Someone would have to do the same thing for windows, i.e. just type in URLs like this:

https://anaconda.org/conda-forge/sparsehash/files

And then check for `win-64` in the file names.

**Comment by count0** [comment]
Are we ready to try this out on graph-tool?

**Comment by stuarteberg** [comment]
We’re still waiting for the following PR to be merged.

https://github.com/conda-forge/sparsehash-feedstock/pull/12

After that, in theory, the bot will open an “Arch Migrator” PR for graph-tool — unless I am mistaken about sparsehash being our last unmigrated dependency.

**Comment by stuarteberg** [comment]
Closed via #97
